internal/cluster/transition.go

```diff
@@ -402,6 +402,10 @@ func (th *transitionHandler) PostRefreshCluster(reason string) stateswitch.PostT
 			err            error
 			updatedCluster *common.Cluster
 		)
+		if sCluster.srcState == *sCluster.cluster.Status && ok && reason == *sCluster.cluster.StatusInfo {
+			return nil
```

Comment:
use swag(*sCluster.cluster.Status) here and then won't need to worry about nulls in the tests and otherwise (see also line 423)

Reply:
L428 is returning `nil`, tho. Doesn't this mean we already should be checking for `nil` in the returned value from this function? Maybe it's worth updating L428 too? 

Let me know if I'm missing something